File: 백준/Silver/11399. ATM/ATM.cs
using System;
using System.IO;

namespace AlgorithmStudy
{
    class boj11399
    {
        public static void Main(string[] args)
        {
            StreamReader sr = new StreamReader(new BufferedStream(Console.OpenStandardInput()));
            int n = int.Parse(sr.ReadLine());
            int[] arr = new int[n];
            string[] input = sr.ReadLine().Split();
            sr.Close();
            for (int i = 0; i < n; i++)
                arr[i] = int.Parse(input[i]);
            Array.Sort(arr);
            for (int i = 1; i < n; i++)
                arr[i] += arr[i - 1];
            int result = 0;
            foreach (int i in arr)
                result += i;
            Console.WriteLine(result);
        }
    }
}

File: 백준/Silver/11399. ATM/ATM.cc
#include <bits/stdc++.h>
using namespace std;

int main() {
	int arr[1000];
	int n;
	cin >> n;
	
	for (int i = 0; i < n; i++)
		cin >> arr[i];

	sort(arr, arr + n);

	for (int i = 1; i < n; i++)
		arr[i] = arr[i - 1] + arr[i];
	int answer = 0;
	for (int i = 0; i < n; i++)
		answer += arr[i];

	printf("%d", answer);

	return 0;
}